src/org/labkey/targetedms/query/SampleFileTable.java

```diff
@@ -171,4 +205,80 @@ public QueryUpdateService getUpdateService()
             }
         };
     }
+
+    private SQLFragment getFileNameSql()
+    {
+        SqlDialect dialect = TargetedMSManager.getSqlDialect();
+
+        // If FilePath is C:\Project1\RawData\sample001.raw
+        // we want sample001.raw
+        // Example SQL on Postgres (sf is alias for SampleFile):
+        // CASE WHEN sf.FilePath IS NOT NULL AND POSITION('\' IN sf.FilePath) > 0
+        //     THEN substr(sf.FilePath, length(sf.FilePath) - POSITION('\' IN  REVERSE(sf.FilePath)) + 2 , length(sf.FilePath))
+        //     ELSE sf.FilePath
+        //     END
+        SQLFragment filePathColSql = new SQLFragment(ExprColumn.STR_TABLE_ALIAS + ".FilePath");
+        SQLFragment filePathLengthSql = new SQLFragment(dialect.getVarcharLengthFunction()).append("(").append(filePathColSql).append(")");
+        return new SQLFragment("CASE")
+                .append(" WHEN ").append(filePathColSql).append(" IS NOT NULL AND ").append(dialect.getStringIndexOfFunction(new SQLFragment("'\\'"), filePathColSql)).append(" > 0")
+                .append(" THEN ").append(dialect.getSubstringFunction(
+                        filePathColSql,
+                        new SQLFragment(filePathLengthSql).append(" - ")
+                                .append(dialect.getStringIndexOfFunction(new SQLFragment("'\\'"), new SQLFragment(" REVERSE(").append(filePathColSql)).append(") + 2 ")),
+                        filePathLengthSql))
+                .append(" ELSE ").append(filePathColSql).append(" END");
+    }
+
+    private class DownloadLinkColumn extends DataColumn
```

Comment:
Should override isSortable() and isFilterable(), since they'll be operating on the underlying column value (a long)

Reply:
Done. Added isSortable() and isFilterable() overrides for the download column of the TargetedMS Runs table as well.